micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -776,6 +778,19 @@ public Config commonTags(String... tags) {
          * @return This configuration instance.
          */
         public synchronized Config meterFilter(MeterFilter filter) {
+            if (!meterMap.isEmpty()) {
+                InternalLogger logger = InternalLoggerFactory.getInstance(MeterRegistry.this.getClass());
+                String baseMessage = "A MeterFilter is being configured after a Meter has been registered to this registry. This is unexpected and should be corrected so all MeterFilters are configured before any Meters are registered. If that is not possible or you have a use case where it should be allowed, contact the Micrometer maintainers.";
```

Comment:
"contact the Micrometer maintainers" is a bit ambiguous what users should do. This was somewhat intentional because whatever logging message we go with will live forever in the versions of Micrometer we release with this, while the way to contact us may change over time. Maybe opening a GitHub issue and providing a link to that will be future-proof enough.

Reply:
Should we say
```suggestion
                String baseMessage = "A MeterFilter is being configured after a Meter has been registered to this registry. This is unexpected and should be corrected so all MeterFilters are configured before any Meters are registered. If that is not possible or you have a use case where it should be allowed, contact the Micrometer maintainers (see: https://micrometer.io).";
```

On the Micrometer landing page, maybe we could have a contact section that groups the slack, github and twitter links together in a drop-down menu item.

Reply:
Linking to the website is less ambiguous but I think it still leaves people wondering which method is best. Maybe any is fine, but that can split the conversation across different places and lose context for everyone. I will open a GitHub issue and include a link to it in the log message. That way we can update the latest info and everyone can read it before reporting to us.

Reply:
I've opened https://github.com/micrometer-metrics/micrometer/issues/4920 for tracking this.